src/DonorProfiles/Profile.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+namespace Give\DonorProfiles;
+
+use \Give_Donor as DonorModel;
+
+class Profile {
+
+	protected $donor;
+	protected $id;
+
+	public function __construct( int $donorId ) {
+		$this->id    = $donorId;
+		$this->donor = new DonorModel( $donorId );
```

Comment:
I would suggest creating a factory that accepts a `$donorId` and then inject the `DonorModel` instance via the constructor, as opposed to creating new objects here.

Reply:
Done, ref: https://github.com/impress-org/givewp/pull/5486/commits/fb9d2b8aa942807130229eae836108dc3715dd69